rust_snuba/src/metrics/global_tags.rs

```diff
@@ -0,0 +1,122 @@
+use std::collections::BTreeMap;
+
+use parking_lot::RwLock;
+use statsdproxy::middleware::Middleware;
+use statsdproxy::types::Metric;
+
+static GLOBAL_TAGS: RwLock<BTreeMap<String, String>> = RwLock::new(BTreeMap::new());
+
+pub fn set_global_tag(key: String, value: String) {
+    GLOBAL_TAGS.write().insert(key, value);
+}
+
+pub struct AddGlobalTags<'a, M> {
+    next: M,
+    global_tags: &'a RwLock<BTreeMap<String, String>>,
+}
+
+impl<M> AddGlobalTags<'static, M>
+where
+    M: Middleware,
+{
+    pub fn new(next: M) -> Self {
+        Self::new_with_tagmap(next, &GLOBAL_TAGS)
+    }
+}
+
+impl<'a, M> AddGlobalTags<'a, M>
+where
+    M: Middleware,
+{
+    pub fn new_with_tagmap(next: M, global_tags: &'a RwLock<BTreeMap<String, String>>) -> Self {
```

Comment:
Are we expecting this function to be used outside of tests?

Reply:
no. i made it private. i can't make it `cfg(test)` though becuase it's still used by `Self::new`

Reply:
You can just use the global version for the tests though, right?

Reply:
my problem is that tests run in parallel, so if I mutate global tags in a test it could bleed into other tests (if they do anything at all with capturing metrics)

Reply:
Ah, I thought each test was its own binary and thus has its own copy of the static. I just looked it up and that is only true for integration tests. TIL!